assert/_diff.ts

```diff
@@ -119,28 +127,28 @@ export function diff<T>(A: T[], B: T[]): Array<DiffResult<T>> {
   }> {
     const M = A.length;
     const N = B.length;
-    const result = [];
+    const result: { type: DiffType; value: T }[] = [];
     let a = M - 1;
     let b = N - 1;
     let j = routes[current.id];
     let type = routes[current.id + diffTypesPtrOffset];
     while (true) {
       if (!j && !type) break;
-      const prev = j;
+      const prev = j!;
       if (type === REMOVED) {
         result.unshift({
           type: swapped ? DiffType.removed : DiffType.added,
-          value: B[b],
+          value: B[b]!,
         });
         b -= 1;
       } else if (type === ADDED) {
         result.unshift({
           type: swapped ? DiffType.added : DiffType.removed,
-          value: A[a],
+          value: A[a]!,
         });
         a -= 1;
       } else {
-        result.unshift({ type: DiffType.common, value: A[a] });
+        result.unshift({ type: DiffType.common, value: A[a]! });
```

Comment:
I feel a bit unsure about this one. We can't say for certain if these indices are correct as it's all dependent on if `routes` was built correctly.

Reply:
I'm unsure what to do here too. @kt3k, WDYT?

Reply:
`diff` is hugely used across Deno ecosystem (via `assertEquals`). I don't think this includes such basic bugs. I think it's fine to use non null assertion here.